<a href="https://colab.research.google.com/github/rommeljpadilla/project2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
#Imports
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


1) Source of data

2) Brief description of data

3) What is the target?

4) Is this a classification or regression problem?

5) How many features?

6) How many rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

In [4]:
dataset_1 = pd.read_csv('/content/drive/MyDrive/Data/all_seasons.csv')
dataset_1.head()


Unnamed: 0      player_name team_abbreviation  age  player_height  \
0           0    Travis Knight               LAL   22         213.36   
1           1        Matt Fish               MIA   27         210.82   
2           2     Matt Bullard               HOU   30         208.28   
3           3     Marty Conlon               BOS   29         210.82   
4           4  Martin Muursepp               DAL   22         205.74   

   player_weight                    college country draft_year draft_round  \
0      106.59412                Connecticut     USA       1996           1   
1      106.59412  North Carolina-Wilmington     USA       1992           2   
2      106.59412                       Iowa     USA  Undrafted   Undrafted   
3      111.13004                 Providence     USA  Undrafted   Undrafted   
4      106.59412                       None     USA       1996           1   

   ...  pts  reb  ast  net_rating  oreb_pct  dreb_pct  usg_pct  ts_pct  \
0  ...  4.8  4.5  0.5         6.2     0.127     0.182    0.142   0.536   
1  ...  0.3  0.8  0.0       -15.1     0.143     0.267    0.265   0.333   
2  ...  4.5  1.6  0.9         0.9     0.016     0.115    0.151   0.535   
3  ...  7.8  4.4  1.4        -9.0     0.083     0.152    0.167   0.542   
4  ...  3.7  1.6  0.5       -14.5     0.109     0.118    0.233   0.482   

   ast_pct   season  
0    0.052  1996-97  
1    0.000  1996-97  
2    0.099  1996-97  
3    0.101  1996-97  
4    0.114  1996-97  

[5 rows x 22 columns]

In [6]:
dataset_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11700 entries, 0 to 11699
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         11700 non-null  int64  
 1   player_name        11700 non-null  object 
 2   team_abbreviation  11700 non-null  object 
 3   age                11700 non-null  int64  
 4   player_height      11700 non-null  float64
 5   player_weight      11700 non-null  float64
 6   college            11700 non-null  object 
 7   country            11700 non-null  object 
 8   draft_year         11700 non-null  object 
 9   draft_round        11700 non-null  object 
 10  draft_number       11700 non-null  object 
 11  gp                 11700 non-null  int64  
 12  pts                11700 non-null  float64
 13  reb                11700 non-null  float64
 14  ast                11700 non-null  float64
 15  net_rating         11700 non-null  float64
 16  oreb_pct           117


1) Source of data
https://www.kaggle.com/justinas/nba-players-data

2) Brief description of data
The data contains years data on each player who has been part of an NBA teams' roster. It captures name,team,age,height,college, country,draft year, draft round, and stats

4) Is this a classification or regression problem?
regression - Can we tell a player's stats based on:
-country
-draft year
-draft round
player_weight
player_height


5) How many features?
5

6) How many rows of data.
11700

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
We are working with 11700 rows;
if an nba player does not get drafted, they appear as 'undrafted' instead of a given year

In [7]:
#source https://www.kaggle.com/danielgrijalvas/movies
dataset_2 = pd.read_csv('/content/drive/MyDrive/Data/movies.csv')
dataset_2.head()

name rating      genre  year  \
0                                     The Shining      R      Drama  1980   
1                                 The Blue Lagoon      R  Adventure  1980   
2  Star Wars: Episode V - The Empire Strikes Back     PG     Action  1980   
3                                       Airplane!     PG     Comedy  1980   
4                                      Caddyshack      R     Comedy  1980   

                        released  score      votes         director  \
0  June 13, 1980 (United States)    8.4   927000.0  Stanley Kubrick   
1   July 2, 1980 (United States)    5.8    65000.0   Randal Kleiser   
2  June 20, 1980 (United States)    8.7  1200000.0   Irvin Kershner   
3   July 2, 1980 (United States)    7.7   221000.0     Jim Abrahams   
4  July 25, 1980 (United States)    7.3   108000.0     Harold Ramis   

                    writer            star         country      budget  \
0             Stephen King  Jack Nicholson  United Kingdom  19000000.0   
1  Henry De Vere Stacpoole  Brooke Shields   United States   4500000.0   
2           Leigh Brackett     Mark Hamill   United States  18000000.0   
3             Jim Abrahams     Robert Hays   United States   3500000.0   
4       Brian Doyle-Murray     Chevy Chase   United States   6000000.0   

         gross             company  runtime  
0   46998772.0        Warner Bros.    146.0  
1   58853106.0   Columbia Pictures    104.0  
2  538375067.0           Lucasfilm    124.0  
3   83453539.0  Paramount Pictures     88.0  
4   39846344.0      Orion Pictures     98.0

In [8]:
dataset_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7668 entries, 0 to 7667
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   name      7668 non-null   object 
 1   rating    7591 non-null   object 
 2   genre     7668 non-null   object 
 3   year      7668 non-null   int64  
 4   released  7666 non-null   object 
 5   score     7665 non-null   float64
 6   votes     7665 non-null   float64
 7   director  7668 non-null   object 
 8   writer    7665 non-null   object 
 9   star      7667 non-null   object 
 10  country   7665 non-null   object 
 11  budget    5497 non-null   float64
 12  gross     7479 non-null   float64
 13  company   7651 non-null   object 
 14  runtime   7664 non-null   float64
dtypes: float64(5), int64(1), object(9)
memory usage: 898.7+ KB



1) Source of data
https://www.kaggle.com/danielgrijalvas/movies

2) Brief description of data
The data contains movie statistics ranging from the previous 30 years

4) Is this a classification or regression problem?
- Can we tell a movie gross based on
-genre
-year released
-budget
-rating
-runtime
-star
-country


5) How many features?
6

6) How many rows of data.
7668

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
There could be so many stars ranging in so many movies.
budget seems like a very important feature but it is the column missing the mmost data
